pkg/sql/sem/builtins/builtins.go

```diff
@@ -2595,6 +2595,113 @@ var builtins = map[string]builtinDefinition{
+	//
+	// PostgreSQL documents date_trunc for text and double precision.
+	// It will also handle smallint, integer, bigint, decimal,
+	// numeric, real, and numeri like text inputs by casting them,
+	// so we support those for compatibility. This gives us the following
+	// function signatures:
+	//
+	//  to_timestamp(text, text)       -> TimestampTZ
+	//  to_timestamp(text)             -> TimestampTZ
+	//  to_timestamp(INT)              -> TimestampTZ
+	//  to_timestamp(INT2)             -> TimestampTZ
+	//  to_timestamp(INT4)             -> TimestampTZ
+	//  to_timestamp(INT8)             -> TimestampTZ
+	//  to_timestamp(FLOAT)            -> TimestampTZ
+	//  to_timestamp(REAL)             -> TimestampTZ
+	//  to_timestamp(DOUBLE PRECISION) -> TimestampTZ
+	//  to_timestamp(DECIMAL)          -> TimestampTZ
+	//
+	// See the following snippet from running the functions in PostgreSQL:
+	//
+	//               postgres=# select to_timestamp(32767::smallint);
+	//               to_timestamp
+	//               ------------------------
+	//               1970-01-01 09:06:07+00
+	//
+	//               postgres=# select to_timestamp(1646906263::integer);
+	//               to_timestamp
+	//               ------------------------
+	//               2022-03-10 09:57:43+00
+	//
+	//               postgres=# select to_timestamp(1646906263::bigint);
+	//               to_timestamp
+	//               ------------------------
+	//               2022-03-10 09:57:43+00
+	//
+	//               postgres=# select to_timestamp(1646906263.123456::decimal);
+	//               to_timestamp
+	//               -------------------------------
+	//               2022-03-10 09:57:43.123456+00
+	//
+	//               postgres=# select to_timestamp(1646906263.123456::numeric);
+	//               to_timestamp
+	//               -------------------------------
+	//               2022-03-10 09:57:43.123456+00
+	//
+	//               postgres=# select to_timestamp(1646906263.123456::real);
+	//               to_timestamp
+	//               ------------------------
+	//               2022-03-10 09:57:20+00
+	//
+	//               postgres=# select to_timestamp('1646906263.123456');
+	//               to_timestamp
+	//               -------------------------------
+	//               2022-03-10 09:57:43.123456+00
+	//
+	"to_timestamp": makeBuiltin(
+		tree.FunctionProperties{Category: categoryDateAndTime},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"timestamp", types.String}},
+			ReturnType: tree.FixedReturnType(types.TimestampTZ),
+			Fn: func(ctx *eval.Context, args tree.Datums) (tree.Datum, error) {
+				ts, err := strconv.ParseFloat(string(tree.MustBeDString(args[0])), 64)
+				if err != nil {
+					return nil, pgerror.New(pgcode.AmbiguousFunction, "invalid input for type text")
+				}
+				return floatToTimestampTZ(ts)
+			},
+			Info:       "Convert Unix epoch (seconds since 1970-01-01 00:00:00+00) to timestamp with time zone.",
+			Volatility: volatility.Immutable,
```

Comment:
I believe this should be stable, not immutable. In Postgres:

```
marcus=# select proargtypes, prorettype::regtype, provolatile, proleakproof from pg_proc where proname = 'to_timestamp';
 proargtypes |        prorettype        | provolatile | proleakproof
-------------+--------------------------+-------------+--------------
 701         | timestamp with time zone | i           | f
 25 25       | timestamp with time zone | s           | f
(2 rows)

marcus=# select 701::oid::regtype, 25::oid::regtype;
     regtype      | regtype
------------------+---------
 double precision | text
(1 row)
```

Reply:
The one arg version is immutable - so this is right right?